Python/optimizer_analysis.c

```diff
@@ -557,7 +558,11 @@ optimize_uops(
         OPT_ERROR_IN_OPCODE(opcode);
     }
     _Py_uop_abstractcontext_fini(ctx);
-    return -1;
+
+    if (PyErr_Occurred()) {
+        PyErr_Clear();
+    }
```

Comment:
Shouldn't this be:

```suggestion
    assert(PyErr_Occurred());
    PyErr_Clear();
```

...or are there cases where an error hasn't occurred, but we end up here?

Reply:
I think the current code goes to error on some cases where we run out of memory too, but no memoryerror is set.

Reply:
I don't think so, all uses of `goto error` in these files seem like they have an error set.